website/blog/2023-11-15-run-webassembly-wasm-workloads-windows-and-macos.md

```diff
@@ -0,0 +1,342 @@
+
+</TabItem>
+<TabItem value="mac" label="macOS">
+On macOS, the support of wasm runtime is not included by default. The podman machine is using Fedora CoreOS and crun/wasm integration is only available in the next stream channel but by default podman machines are using testing channel.
+
+- Option 1: use next channel of Fedora CoreOS
+  You need to create a compliant machine first by running podman machine init command using next as image-path so it'll fetch the next channel Fedora CoreOS instead of the latest testing binaries.
+
+  Here is the command to create and start a new podman machine including WASM:
+
+  ```shell-session
+  $ podman machine init --cpus 2 --memory 2048 --image-path next --now
+  ```
+
+  You can use another machine by adding a parameter (machine name) at the end of the command line:
+
+  ```shell-session
+  $ podman machine init --cpus 2 --memory 2048 --image-path next --now wasm
+  ```
+
+- Option 2: install dependencies in the current Fedora CoreOS machine.
+
+  It's also possible to connect to the podman machine with podman machine ssh command and then install the wasm runtime using
+
+  ```shell-session
+  $ sudo rpm-ostree install crun-wasm wasmedge-rt
+  ```
+
+</TabItem>
+</Tabs>
+ 
+
+### Running WASM images
+
+Let's try with a simple hello world sample.
+
+We will use example coming from https://github.com/redhat-developer/podman-desktop-demo/tree/main/wasm/rust-hello-world
+
+There is already an OCI image on quay.io
+
+To run the workload, we will use the following command:
+
+```shell-shession
+$ podman run --platform wasi/wasm quay.io/podman-desktop-demo/wasm-rust-hello-world
+```
+
+When running the command, you will see a Podman Hello World that was compiled using a Rust project using the println function and compiled into WASM using `--target wasm32-wasi` parameter at compilation time.
+
+![Hello World example running](img/wasm-workloads-on-macos-and-windows-with-podman/wasm-hello-world.png)
+
+you can omit the `--platform wasi/wasm` flag but in that case you'll get a warning that the platform of the image is not matching the platform of your computer (`WARNING: image platform (wasi/wasm) does not match the expected platform (linux/arm64)`)
+
+From this point, you can run other OCI images using wasm workloads, not only the podman hello world sample.
+
+**_NOTE:_** if you don't have the prerequisites installed in your podman machine you will see this error: `Error: requested OCI runtime crun-wasm is not available: invalid argument`
+
+In that case you should check that the prerequisites from the previous section are met.
+
+## Building wasm OCI images with podman
+
+### Building with a specific platform/architecture
+
+Running wasm workload is an interesting use case from a consumer point of view. It helps to consume wasm binaries. But another interesting case is to distribute and build these wasm images so anyone could run them quickly.
+
```

Comment:
I think you should talk earlier that you are working with a multi-stage Dockerfile (the first - Build - stage in a std platform, and the second - Run - stage in wasm), or it is difficult to follow

Reply:
ok

Reply:
Added a line with multi-stage build